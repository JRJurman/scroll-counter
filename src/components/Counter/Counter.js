import { registerHtml } from 'tram-one'
import ScrollCounter from '../ScrollCounter'
import CounterLabel from '../CounterLabel'
import ScrollCover from '../ScrollCover'
import './Counter.scss'

const html = registerHtml({
  CounterLabel,
  ScrollCounter,
  ScrollCover
})

export default ({ id, color }) => {
  const counterStyle = `
    background: ${color};
  `
  return html`
    <div class="Counter" style=${counterStyle}>
      <CounterLabel counterId=${id}/>
      <ScrollCounter counterId=${id}/>
      <ScrollCover color=${color} />
    </div>
  `
}
