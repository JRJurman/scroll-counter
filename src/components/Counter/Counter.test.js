import Counter from './Counter'

describe('Counter', () => {
  describe('with default props', () => {
    const wrapper = Counter({ id: 0, color: '#FFEEDD' })

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
