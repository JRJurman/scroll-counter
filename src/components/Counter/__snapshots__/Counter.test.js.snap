// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter with default props should match snapshot 1`] = `
"<div style=\\"
    background: #FFEEDD;
  \\" class=\\"Counter\\"><input placeholder=\\"Counter\\" value=\\"\\" class=\\"CounterLabel\\"><div class=\\"ScrollCounter\\"><!-- padding before scrollable elements --><div counter-id=\\"0\\" value=\\"\\" class=\\"ScrollElement\\"></div><div counter-id=\\"0\\" value=\\"\\" class=\\"ScrollElement\\"></div><!-- scrollable elements --><div counter-id=\\"0\\" value=\\"-100\\" class=\\"ScrollElement\\">-100</div><div counter-id=\\"0\\" value=\\"-99\\" class=\\"ScrollElement\\">-99</div><div counter-id=\\"0\\" value=\\"-98\\" class=\\"ScrollElement\\">-98</div><div counter-id=\\"0\\" value=\\"-97\\" class=\\"ScrollElement\\">-97</div><div counter-id=\\"0\\" value=\\"-96\\" class=\\"ScrollElement\\">-96</div><div counter-id=\\"0\\" value=\\"-95\\" class=\\"ScrollElement\\">-95</div><div counter-id=\\"0\\" value=\\"-94\\" class=\\"ScrollElement\\">-94</div><div counter-id=\\"0\\" value=\\"-93\\" class=\\"ScrollElement\\">-93</div><div counter-id=\\"0\\" value=\\"-92\\" class=\\"ScrollElement\\">-92</div><div counter-id=\\"0\\" value=\\"-91\\" class=\\"ScrollElement\\">-91</div><div counter-id=\\"0\\" value=\\"-90\\" class=\\"ScrollElement\\">-90</div><div counter-id=\\"0\\" value=\\"-89\\" class=\\"ScrollElement\\">-89</div><div counter-id=\\"0\\" value=\\"-88\\" class=\\"ScrollElement\\">-88</div><div counter-id=\\"0\\" value=\\"-87\\" class=\\"ScrollElement\\">-87</div><div counter-id=\\"0\\" value=\\"-86\\" class=\\"ScrollElement\\">-86</div><div counter-id=\\"0\\" value=\\"-85\\" class=\\"ScrollElement\\">-85</div><div counter-id=\\"0\\" value=\\"-84\\" class=\\"ScrollElement\\">-84</div><div counter-id=\\"0\\" value=\\"-83\\" class=\\"ScrollElement\\">-83</div><div counter-id=\\"0\\" value=\\"-82\\" class=\\"ScrollElement\\">-82</div><div counter-id=\\"0\\" value=\\"-81\\" class=\\"ScrollElement\\">-81</div><div counter-id=\\"0\\" value=\\"-80\\" class=\\"ScrollElement\\">-80</div><div counter-id=\\"0\\" value=\\"-79\\" class=\\"ScrollElement\\">-79</div><div counter-id=\\"0\\" value=\\"-78\\" class=\\"ScrollElement\\">-78</div><div counter-id=\\"0\\" value=\\"-77\\" class=\\"ScrollElement\\">-77</div><div counter-id=\\"0\\" value=\\"-76\\" class=\\"ScrollElement\\">-76</div><div counter-id=\\"0\\" value=\\"-75\\" class=\\"ScrollElement\\">-75</div><div counter-id=\\"0\\" value=\\"-74\\" class=\\"ScrollElement\\">-74</div><div counter-id=\\"0\\" value=\\"-73\\" class=\\"ScrollElement\\">-73</div><div counter-id=\\"0\\" value=\\"-72\\" class=\\"ScrollElement\\">-72</div><div counter-id=\\"0\\" value=\\"-71\\" class=\\"ScrollElement\\">-71</div><div counter-id=\\"0\\" value=\\"-70\\" class=\\"ScrollElement\\">-70</div><div counter-id=\\"0\\" value=\\"-69\\" class=\\"ScrollElement\\">-69</div><div counter-id=\\"0\\" value=\\"-68\\" class=\\"ScrollElement\\">-68</div><div counter-id=\\"0\\" value=\\"-67\\" class=\\"ScrollElement\\">-67</div><div counter-id=\\"0\\" value=\\"-66\\" class=\\"ScrollElement\\">-66</div><div counter-id=\\"0\\" value=\\"-65\\" class=\\"ScrollElement\\">-65</div><div counter-id=\\"0\\" value=\\"-64\\" class=\\"ScrollElement\\">-64</div><div counter-id=\\"0\\" value=\\"-63\\" class=\\"ScrollElement\\">-63</div><div counter-id=\\"0\\" value=\\"-62\\" class=\\"ScrollElement\\">-62</div><div counter-id=\\"0\\" value=\\"-61\\" class=\\"ScrollElement\\">-61</div><div counter-id=\\"0\\" value=\\"-60\\" class=\\"ScrollElement\\">-60</div><div counter-id=\\"0\\" value=\\"-59\\" class=\\"ScrollElement\\">-59</div><div counter-id=\\"0\\" value=\\"-58\\" class=\\"ScrollElement\\">-58</div><div counter-id=\\"0\\" value=\\"-57\\" class=\\"ScrollElement\\">-57</div><div counter-id=\\"0\\" value=\\"-56\\" class=\\"ScrollElement\\">-56</div><div counter-id=\\"0\\" value=\\"-55\\" class=\\"ScrollElement\\">-55</div><div counter-id=\\"0\\" value=\\"-54\\" class=\\"ScrollElement\\">-54</div><div counter-id=\\"0\\" value=\\"-53\\" class=\\"ScrollElement\\">-53</div><div counter-id=\\"0\\" value=\\"-52\\" class=\\"ScrollElement\\">-52</div><div counter-id=\\"0\\" value=\\"-51\\" class=\\"ScrollElement\\">-51</div><div counter-id=\\"0\\" value=\\"-50\\" class=\\"ScrollElement\\">-50</div><div counter-id=\\"0\\" value=\\"-49\\" class=\\"ScrollElement\\">-49</div><div counter-id=\\"0\\" value=\\"-48\\" class=\\"ScrollElement\\">-48</div><div counter-id=\\"0\\" value=\\"-47\\" class=\\"ScrollElement\\">-47</div><div counter-id=\\"0\\" value=\\"-46\\" class=\\"ScrollElement\\">-46</div><div counter-id=\\"0\\" value=\\"-45\\" class=\\"ScrollElement\\">-45</div><div counter-id=\\"0\\" value=\\"-44\\" class=\\"ScrollElement\\">-44</div><div counter-id=\\"0\\" value=\\"-43\\" class=\\"ScrollElement\\">-43</div><div counter-id=\\"0\\" value=\\"-42\\" class=\\"ScrollElement\\">-42</div><div counter-id=\\"0\\" value=\\"-41\\" class=\\"ScrollElement\\">-41</div><div counter-id=\\"0\\" value=\\"-40\\" class=\\"ScrollElement\\">-40</div><div counter-id=\\"0\\" value=\\"-39\\" class=\\"ScrollElement\\">-39</div><div counter-id=\\"0\\" value=\\"-38\\" class=\\"ScrollElement\\">-38</div><div counter-id=\\"0\\" value=\\"-37\\" class=\\"ScrollElement\\">-37</div><div counter-id=\\"0\\" value=\\"-36\\" class=\\"ScrollElement\\">-36</div><div counter-id=\\"0\\" value=\\"-35\\" class=\\"ScrollElement\\">-35</div><div counter-id=\\"0\\" value=\\"-34\\" class=\\"ScrollElement\\">-34</div><div counter-id=\\"0\\" value=\\"-33\\" class=\\"ScrollElement\\">-33</div><div counter-id=\\"0\\" value=\\"-32\\" class=\\"ScrollElement\\">-32</div><div counter-id=\\"0\\" value=\\"-31\\" class=\\"ScrollElement\\">-31</div><div counter-id=\\"0\\" value=\\"-30\\" class=\\"ScrollElement\\">-30</div><div counter-id=\\"0\\" value=\\"-29\\" class=\\"ScrollElement\\">-29</div><div counter-id=\\"0\\" value=\\"-28\\" class=\\"ScrollElement\\">-28</div><div counter-id=\\"0\\" value=\\"-27\\" class=\\"ScrollElement\\">-27</div><div counter-id=\\"0\\" value=\\"-26\\" class=\\"ScrollElement\\">-26</div><div counter-id=\\"0\\" value=\\"-25\\" class=\\"ScrollElement\\">-25</div><div counter-id=\\"0\\" value=\\"-24\\" class=\\"ScrollElement\\">-24</div><div counter-id=\\"0\\" value=\\"-23\\" class=\\"ScrollElement\\">-23</div><div counter-id=\\"0\\" value=\\"-22\\" class=\\"ScrollElement\\">-22</div><div counter-id=\\"0\\" value=\\"-21\\" class=\\"ScrollElement\\">-21</div><div counter-id=\\"0\\" value=\\"-20\\" class=\\"ScrollElement\\">-20</div><div counter-id=\\"0\\" value=\\"-19\\" class=\\"ScrollElement\\">-19</div><div counter-id=\\"0\\" value=\\"-18\\" class=\\"ScrollElement\\">-18</div><div counter-id=\\"0\\" value=\\"-17\\" class=\\"ScrollElement\\">-17</div><div counter-id=\\"0\\" value=\\"-16\\" class=\\"ScrollElement\\">-16</div><div counter-id=\\"0\\" value=\\"-15\\" class=\\"ScrollElement\\">-15</div><div counter-id=\\"0\\" value=\\"-14\\" class=\\"ScrollElement\\">-14</div><div counter-id=\\"0\\" value=\\"-13\\" class=\\"ScrollElement\\">-13</div><div counter-id=\\"0\\" value=\\"-12\\" class=\\"ScrollElement\\">-12</div><div counter-id=\\"0\\" value=\\"-11\\" class=\\"ScrollElement\\">-11</div><div counter-id=\\"0\\" value=\\"-10\\" class=\\"ScrollElement\\">-10</div><div counter-id=\\"0\\" value=\\"-9\\" class=\\"ScrollElement\\">-9</div><div counter-id=\\"0\\" value=\\"-8\\" class=\\"ScrollElement\\">-8</div><div counter-id=\\"0\\" value=\\"-7\\" class=\\"ScrollElement\\">-7</div><div counter-id=\\"0\\" value=\\"-6\\" class=\\"ScrollElement\\">-6</div><div counter-id=\\"0\\" value=\\"-5\\" class=\\"ScrollElement\\">-5</div><div counter-id=\\"0\\" value=\\"-4\\" class=\\"ScrollElement\\">-4</div><div counter-id=\\"0\\" value=\\"-3\\" class=\\"ScrollElement\\">-3</div><div counter-id=\\"0\\" value=\\"-2\\" class=\\"ScrollElement\\">-2</div><div counter-id=\\"0\\" value=\\"-1\\" class=\\"ScrollElement\\">-1</div><div counter-id=\\"0\\" value=\\"0\\" class=\\"ScrollElement\\">0</div><div counter-id=\\"0\\" value=\\"1\\" class=\\"ScrollElement\\">1</div><div counter-id=\\"0\\" value=\\"2\\" class=\\"ScrollElement\\">2</div><div counter-id=\\"0\\" value=\\"3\\" class=\\"ScrollElement\\">3</div><div counter-id=\\"0\\" value=\\"4\\" class=\\"ScrollElement\\">4</div><div counter-id=\\"0\\" value=\\"5\\" class=\\"ScrollElement\\">5</div><div counter-id=\\"0\\" value=\\"6\\" class=\\"ScrollElement\\">6</div><div counter-id=\\"0\\" value=\\"7\\" class=\\"ScrollElement\\">7</div><div counter-id=\\"0\\" value=\\"8\\" class=\\"ScrollElement\\">8</div><div counter-id=\\"0\\" value=\\"9\\" class=\\"ScrollElement\\">9</div><div counter-id=\\"0\\" value=\\"10\\" class=\\"ScrollElement\\">10</div><div counter-id=\\"0\\" value=\\"11\\" class=\\"ScrollElement\\">11</div><div counter-id=\\"0\\" value=\\"12\\" class=\\"ScrollElement\\">12</div><div counter-id=\\"0\\" value=\\"13\\" class=\\"ScrollElement\\">13</div><div counter-id=\\"0\\" value=\\"14\\" class=\\"ScrollElement\\">14</div><div counter-id=\\"0\\" value=\\"15\\" class=\\"ScrollElement\\">15</div><div counter-id=\\"0\\" value=\\"16\\" class=\\"ScrollElement\\">16</div><div counter-id=\\"0\\" value=\\"17\\" class=\\"ScrollElement\\">17</div><div counter-id=\\"0\\" value=\\"18\\" class=\\"ScrollElement\\">18</div><div counter-id=\\"0\\" value=\\"19\\" class=\\"ScrollElement\\">19</div><div counter-id=\\"0\\" value=\\"20\\" class=\\"ScrollElement\\">20</div><div counter-id=\\"0\\" value=\\"21\\" class=\\"ScrollElement\\">21</div><div counter-id=\\"0\\" value=\\"22\\" class=\\"ScrollElement\\">22</div><div counter-id=\\"0\\" value=\\"23\\" class=\\"ScrollElement\\">23</div><div counter-id=\\"0\\" value=\\"24\\" class=\\"ScrollElement\\">24</div><div counter-id=\\"0\\" value=\\"25\\" class=\\"ScrollElement\\">25</div><div counter-id=\\"0\\" value=\\"26\\" class=\\"ScrollElement\\">26</div><div counter-id=\\"0\\" value=\\"27\\" class=\\"ScrollElement\\">27</div><div counter-id=\\"0\\" value=\\"28\\" class=\\"ScrollElement\\">28</div><div counter-id=\\"0\\" value=\\"29\\" class=\\"ScrollElement\\">29</div><div counter-id=\\"0\\" value=\\"30\\" class=\\"ScrollElement\\">30</div><div counter-id=\\"0\\" value=\\"31\\" class=\\"ScrollElement\\">31</div><div counter-id=\\"0\\" value=\\"32\\" class=\\"ScrollElement\\">32</div><div counter-id=\\"0\\" value=\\"33\\" class=\\"ScrollElement\\">33</div><div counter-id=\\"0\\" value=\\"34\\" class=\\"ScrollElement\\">34</div><div counter-id=\\"0\\" value=\\"35\\" class=\\"ScrollElement\\">35</div><div counter-id=\\"0\\" value=\\"36\\" class=\\"ScrollElement\\">36</div><div counter-id=\\"0\\" value=\\"37\\" class=\\"ScrollElement\\">37</div><div counter-id=\\"0\\" value=\\"38\\" class=\\"ScrollElement\\">38</div><div counter-id=\\"0\\" value=\\"39\\" class=\\"ScrollElement\\">39</div><div counter-id=\\"0\\" value=\\"40\\" class=\\"ScrollElement\\">40</div><div counter-id=\\"0\\" value=\\"41\\" class=\\"ScrollElement\\">41</div><div counter-id=\\"0\\" value=\\"42\\" class=\\"ScrollElement\\">42</div><div counter-id=\\"0\\" value=\\"43\\" class=\\"ScrollElement\\">43</div><div counter-id=\\"0\\" value=\\"44\\" class=\\"ScrollElement\\">44</div><div counter-id=\\"0\\" value=\\"45\\" class=\\"ScrollElement\\">45</div><div counter-id=\\"0\\" value=\\"46\\" class=\\"ScrollElement\\">46</div><div counter-id=\\"0\\" value=\\"47\\" class=\\"ScrollElement\\">47</div><div counter-id=\\"0\\" value=\\"48\\" class=\\"ScrollElement\\">48</div><div counter-id=\\"0\\" value=\\"49\\" class=\\"ScrollElement\\">49</div><div counter-id=\\"0\\" value=\\"50\\" class=\\"ScrollElement\\">50</div><div counter-id=\\"0\\" value=\\"51\\" class=\\"ScrollElement\\">51</div><div counter-id=\\"0\\" value=\\"52\\" class=\\"ScrollElement\\">52</div><div counter-id=\\"0\\" value=\\"53\\" class=\\"ScrollElement\\">53</div><div counter-id=\\"0\\" value=\\"54\\" class=\\"ScrollElement\\">54</div><div counter-id=\\"0\\" value=\\"55\\" class=\\"ScrollElement\\">55</div><div counter-id=\\"0\\" value=\\"56\\" class=\\"ScrollElement\\">56</div><div counter-id=\\"0\\" value=\\"57\\" class=\\"ScrollElement\\">57</div><div counter-id=\\"0\\" value=\\"58\\" class=\\"ScrollElement\\">58</div><div counter-id=\\"0\\" value=\\"59\\" class=\\"ScrollElement\\">59</div><div counter-id=\\"0\\" value=\\"60\\" class=\\"ScrollElement\\">60</div><div counter-id=\\"0\\" value=\\"61\\" class=\\"ScrollElement\\">61</div><div counter-id=\\"0\\" value=\\"62\\" class=\\"ScrollElement\\">62</div><div counter-id=\\"0\\" value=\\"63\\" class=\\"ScrollElement\\">63</div><div counter-id=\\"0\\" value=\\"64\\" class=\\"ScrollElement\\">64</div><div counter-id=\\"0\\" value=\\"65\\" class=\\"ScrollElement\\">65</div><div counter-id=\\"0\\" value=\\"66\\" class=\\"ScrollElement\\">66</div><div counter-id=\\"0\\" value=\\"67\\" class=\\"ScrollElement\\">67</div><div counter-id=\\"0\\" value=\\"68\\" class=\\"ScrollElement\\">68</div><div counter-id=\\"0\\" value=\\"69\\" class=\\"ScrollElement\\">69</div><div counter-id=\\"0\\" value=\\"70\\" class=\\"ScrollElement\\">70</div><div counter-id=\\"0\\" value=\\"71\\" class=\\"ScrollElement\\">71</div><div counter-id=\\"0\\" value=\\"72\\" class=\\"ScrollElement\\">72</div><div counter-id=\\"0\\" value=\\"73\\" class=\\"ScrollElement\\">73</div><div counter-id=\\"0\\" value=\\"74\\" class=\\"ScrollElement\\">74</div><div counter-id=\\"0\\" value=\\"75\\" class=\\"ScrollElement\\">75</div><div counter-id=\\"0\\" value=\\"76\\" class=\\"ScrollElement\\">76</div><div counter-id=\\"0\\" value=\\"77\\" class=\\"ScrollElement\\">77</div><div counter-id=\\"0\\" value=\\"78\\" class=\\"ScrollElement\\">78</div><div counter-id=\\"0\\" value=\\"79\\" class=\\"ScrollElement\\">79</div><div counter-id=\\"0\\" value=\\"80\\" class=\\"ScrollElement\\">80</div><div counter-id=\\"0\\" value=\\"81\\" class=\\"ScrollElement\\">81</div><div counter-id=\\"0\\" value=\\"82\\" class=\\"ScrollElement\\">82</div><div counter-id=\\"0\\" value=\\"83\\" class=\\"ScrollElement\\">83</div><div counter-id=\\"0\\" value=\\"84\\" class=\\"ScrollElement\\">84</div><div counter-id=\\"0\\" value=\\"85\\" class=\\"ScrollElement\\">85</div><div counter-id=\\"0\\" value=\\"86\\" class=\\"ScrollElement\\">86</div><div counter-id=\\"0\\" value=\\"87\\" class=\\"ScrollElement\\">87</div><div counter-id=\\"0\\" value=\\"88\\" class=\\"ScrollElement\\">88</div><div counter-id=\\"0\\" value=\\"89\\" class=\\"ScrollElement\\">89</div><div counter-id=\\"0\\" value=\\"90\\" class=\\"ScrollElement\\">90</div><div counter-id=\\"0\\" value=\\"91\\" class=\\"ScrollElement\\">91</div><div counter-id=\\"0\\" value=\\"92\\" class=\\"ScrollElement\\">92</div><div counter-id=\\"0\\" value=\\"93\\" class=\\"ScrollElement\\">93</div><div counter-id=\\"0\\" value=\\"94\\" class=\\"ScrollElement\\">94</div><div counter-id=\\"0\\" value=\\"95\\" class=\\"ScrollElement\\">95</div><div counter-id=\\"0\\" value=\\"96\\" class=\\"ScrollElement\\">96</div><div counter-id=\\"0\\" value=\\"97\\" class=\\"ScrollElement\\">97</div><div counter-id=\\"0\\" value=\\"98\\" class=\\"ScrollElement\\">98</div><div counter-id=\\"0\\" value=\\"99\\" class=\\"ScrollElement\\">99</div><div counter-id=\\"0\\" value=\\"100\\" class=\\"ScrollElement\\">100</div><!-- padding following scrollable elements --><div counter-id=\\"0\\" value=\\"\\" class=\\"ScrollElement\\"></div><div counter-id=\\"0\\" value=\\"\\" class=\\"ScrollElement\\"></div></div><div style=\\"
    background: linear-gradient(to right, #FFEEDDC9 30%, transparent 40%, transparent 60%, #FFEEDDC9 70%);
  \\" class=\\"ScrollCover\\"></div></div>"
`;
